src/global-state/types.ts

```diff
@@ -1,17 +1,34 @@
+import { MutableRefObject } from "react";
 import {
   TJoinProductionOptions,
   TProduction,
 } from "../components/production-line/types.ts";
 
-export type TGlobalState = {
+export interface ErrorState {
+  globalError?: Error | null;
+  callErrors?: Record<string, Error> | null;
+}
+
+export interface CallState {
+  id: string;
+  peerConnection: MutableRefObject<RTCPeerConnection | null> | null;
```

Comment:
there should be no need for this to be a ref object, I believe it is dangerous in the context of managing the peer connections and can easily introduce unexpected bugs